tools/its.py

```diff
@@ -289,6 +291,132 @@ def get_chromedriver(self, headless):
+            target_y (int): y coordinate of target
+
+        Returns:
+            tuple: List of waypoints (x_coordinates, y_coordinates)
+        """
+        
+        # Init and add source coordinates
+        x_coordinates = [source_x]
+        y_coordinates = [source_y]
+        
+        x_target_reached = False
+        y_target_reached = False
+        
+        while not x_target_reached or not y_target_reached:
+            
+            # Create new random waypoints
+            source_x = source_x + randint(3,50)
+            source_y = source_y + randint(3,50)
+            
+            # If targets reached, stay at target
+            if source_x >= target_x:
+                source_x = target_x
+                x_target_reached = True
+            if source_y >= target_y:
+                source_y = target_y
+                y_target_reached = True
+            
+            # Append new waypoint coordinates
+            x_coordinates.append(source_x)
+            y_coordinates.append(source_y)
+                
+        return (x_coordinates, y_coordinates)
+
+    def move_mouse_by_offsets(self, x_coordinates: list, y_coordinates: list, driver) -> tuple:
+        """Move mouse by offeset to list of x and y coordinates
+
+        Args:
+            x_coordinates (list): x waypoints
+            y_coordinates (list): y waypoints
+            driver : Chromedriver
+
+        Returns:
+            tuple: Current mouse coordinates (mouse_x, mouse_y)
+        """
+        
+        # Get current window size
+        window_width = driver.get_window_size()["width"]
+        window_height = driver.get_window_size()["height"]
+        
+        # Init current mouse position
+        current_mouse_x = 0
+        current_mouse_y = 0
+
+        # Append mouse movement for each x,y coordinate
+        for index, coordinate in enumerate(zip(x_coordinates, y_coordinates)):
+
+            # Get current window size
+            window_width = driver.get_window_size()["width"]
+            window_height = driver.get_window_size()["height"]
+
+            # If not first index calculate difference to last coordinate and update next offset
+            if index:
+                x_offset = coordinate[0] - x_coordinates[index - 1]
+                y_offset = coordinate[1] - y_coordinates[index - 1]
+            # First offset has no previous coordinate
+            else:
+                x_offset = coordinate[0]
+                y_offset = coordinate[1]
+            
+            # Update predicted mouse position
+            current_mouse_x = current_mouse_x + x_offset
+            current_mouse_y = current_mouse_y + y_offset
+            
+            # Check wether predicted mouse position is out of bounds
+            if current_mouse_x >= window_width or current_mouse_y >= window_height:
+                #print(f"OUT OF BOUNDS offset: x:{x_offset} , y:{y_offset} not added\n")
+                pass
+            else:
```

Comment:
würde das umformulieren in ein `if not (...): ...` anstatt `if: pass else: ...`

Reply:
-> if not current_mouse_x >= window_width and not current_mouse_y >= window_height: